xgcm/grid.py

```diff
@@ -352,43 +354,74 @@ def face_edge_data(fnum, face_axis, count=1):
             edge_data = concatenate(arrays, face_axis_num)
         else:
             edge_data = face_edge_data(None, None)
-
+        if self._periodic:
+            if boundary_discontinuity:
+                if is_left_edge:
+                    edge_data = edge_data - boundary_discontinuity
+                elif not is_left_edge:
+                    edge_data = edge_data + boundary_discontinuity
+
+        # Check consistency of output data
+        if not isinstance(edge_data, type(da.data)):
+            raise RuntimeError('Types do not match edge(%s)/da.data(%s)'
+                               % (type(edge_data), type(da.data)))
         return edge_data
 
-
     def _extend_left(self, da, boundary=None, fill_value=0.0,
-                     ignore_connections=False, vector_partner=None):
+                     ignore_connections=False, vector_partner=None,
+                     boundary_discontinuity=None):
+        if not isinstance(da, xr.DataArray):
+            raise RuntimeError('Input `da` needs to be a xarray.DataArray. \
+                                Is:%s' % (type(da)))
```

Comment:
I would prefer not to do type checking on private methods. Since they are private (starting with underscore), we can assume they will not be misused. Where in our code would this error be raised?